### PR TITLE
feat(scala): Added ScalaProvider

### LINKED ISSUE #714: Scala sbt support
### Feature request

Provide support for Scala sbt projects to compile to jar files that can be deployed similarly to the way maven and gradle is handled

### Motivation

I want to build docker images for my sbt application while currently only maven and gradle are supported.

### Contribution

Would love to help by sending a PR

### PR DESCRIPTION
**Pull request by yzia2000**
Fixes #714

The ScalaProvider adds support for the scala ecosystem, not covered by maven and gradle which includes sbt, mill and scala-cli.

Added support for sbt projects.

### CODE REVIEW

**Comment by yzia2000** [comment]
Hi this PR adds support for sbt projects. In the future, mill and scala-cli will be added as well. This should support a wide range of the community which mainly use sbt, mill and scala-cli (native to scala ecosystem) and maven, gradle which are already covered under java

File: src/providers/scala.rs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    },
+};
+use anyhow::Result;
+
+pub struct ScalaProvider {}
+
+/**
+ * Scala provider currently supports sbt.
+ * - The sbt project requires sbt-native-packager, a popular packaging
+ *   tool used by the community to package the. Setting executableScriptName and
```
Comment (coffee-cup) [inline src/providers/scala.rs:18]: Feel free to ignore suggestion and use something else, but the comment is missing something

```suggestion
 *   tool used by the community to package apps. Setting executableScriptName and
```

**Review by coffee-cup — COMMENTED** [review]
Thanks for the PR!
This is looking good so far. Can you please update the snapshot tests https://github.com/railwayapp/nixpacks/blob/main/CONTRIBUTING.md#snapshot-tests

**Comment by coffee-cup** [comment]
Ended up quickly adding the snapshot and fixing linting. Thanks again for the PR 🎉

**Comment by yzia2000** [comment]
> Ended up quickly adding the snapshot and fixing linting. Thanks again for the PR 🎉 

Ah that's a real pity. I live in an asian timezone so i was asleep during the period. Thanks!

**Comment by yzia2000** [comment]
Also @coffee-cup on a sidenote. Do you know when releases happen. When can I expect to use scala SBT on railway?

**Pull request merged**